pkg/operator/controller/externaldns/deployment.go

```diff
@@ -372,10 +372,11 @@ func externalDNSContainersChanged(current, expected, updated *appsv1.Deployment)
 	currentContMap := buildIndexedContainerMap(current.Spec.Template.Spec.Containers)
 	expectedContMap := buildIndexedContainerMap(expected.Spec.Template.Spec.Containers)
 
-	// let's check that all the current containers have the desired values set
-	for currName, currCont := range currentContMap {
-		// if the current container is expected: check its fields
-		if expCont, found := expectedContMap[currName]; found {
+	// ensure all expected containers are present,
```

Comment:
This looks good, but have you considered using a deep copy instead of assignment in line 368, where you reset them all?

Reply:
Yes, we could do a deep copy, that would be a little more cpu cycles but a bit safer in terms of the expected containers modifications done via `updated` variable by mistake. But here, we were safe as 1) no other function touches containers and 2) we return immediately so the same function doesn't have a chance to modify `updated` variable.

However your comment made me think about the `if` condition which you pointed into. Actually it prevents the extra containers from being kept in case the lengths of the current `pod.spec.containers` is not the same as expected, so in a lot of cases. I removed this condition to enable the injection of the extra container(s) in any case.

Reply:
`externalDNSDeploymentChanged` already makes a deep copy of `current` to make `update`, so it seems to me that it would be superfluous to do another deep copy here.